File: code/library/src/main/java/org/fbme/lib/iec61499/fbnetwork/subapp/SubapplicationDeclaration.java
package org.fbme.lib.iec61499.fbnetwork.subapp;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Reference;
import org.fbme.lib.iec61499.declarations.SubapplicationTypeDeclaration;
import org.fbme.lib.iec61499.fbnetwork.FunctionBlockDeclarationBase;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface SubapplicationDeclaration extends FunctionBlockDeclarationBase {

    @Nullable SubappNetwork getContainer();

    @NotNull Reference<SubapplicationTypeDeclaration> getTypeReference();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/fbnetwork/subapp/SubapplicationDeclaration.kt
package org.fbme.lib.iec61499.fbnetwork.subapp

import org.fbme.lib.common.Reference
import org.fbme.lib.iec61499.declarations.SubapplicationTypeDeclaration
import org.fbme.lib.iec61499.fbnetwork.FunctionBlockDeclarationBase

interface SubapplicationDeclaration : FunctionBlockDeclarationBase {
    override val container: SubappNetwork?
    val typeReference: Reference<SubapplicationTypeDeclaration>
}
